### ISSUE TITLE
YouTube ReVanced will crash at the beginning of regular video playback due to missing resource for notification icon.

### ISSUE THREAD

**Issue by ankit1w** [j-hc/revanced-magisk-module#246]
Full details on this issue : https://github.com/revanced/revanced-patches/issues/2291

**Comment by j-hc** [comment]
Does it happen with magisk module or apk or both

**Comment by ramanveerji** [comment]
> Does it happen with magisk module or apk or both

No crash, working fine for me.

**Comment by ankit1w** [comment]
> Does it happen with magisk module or apk or both

Just the module

**Comment by ankit1w** [comment]
> what gapps do you use? does it come with youtube included?

I use this ROM https://t.me/RedmiNote10ProChannel/1370, gapps came preinstalled

No, it doesn't come with YouTube included

**Comment by TheNeutrinoRaged** [comment]
I had this issue with the magisk module on lineage20 on poco F1. I used mindthegapps which does not include youtube

apk works fine for me

**Comment by TheNeutrinoRaged** [comment]
Just tried the module again, same crash with yesterday's release. Below is the logcat entries from the crash - playback started fine continued in pip when I minimised youtube but background play triggered the crash.

```
06-01 17:02:16.773  2851  2851 E Icon    : Unable to load resource 0x7f080a59 from pkg=com.google.android.youtube
06-01 17:02:18.177  5285  5285 E AndroidRuntime: Process: com.google.android.youtube, PID: 5285
06-01 17:02:18.177  5285  5285 E AndroidRuntime: android.app.RemoteServiceException$BadForegroundServiceNotificationException: Bad notification(tag=null, id=2) posted from package com.google.android.youtube, crashing app(uid=10299, pid=5285): Couldn't create icon StatusBarIcon(icon=Icon(typ=RESOURCE pkg=com.google.android.youtube id=0x7f080a59) visible user=0 )
06-01 17:02:18.287  2851  2851 E Icon    : Unable to load resource 0x7f080a59 from pkg=com.google.android.youtube
06-01 17:02:18.317  2851  2851 E Icon    : Unable to load resource 0x7f080a59 from pkg=com.google.android.youtube
06-01 17:02:18.636  2851  2851 E Icon    : Unable to load resource 0x7f080a59 from pkg=com.google.android.youtube
06-01 17:02:18.636  2851  2851 E Icon    : Unable to load resource 0x7f080a59 from pkg=com.google.android.youtube
```

**Comment by yanhuasanyuexyz** [comment]
Also magisk module will crash

**Comment by j-hc** [comment]
test this module:
https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb

**Comment by yanhuasanyuexyz** [comment]
> test this module: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb

It still doesn't work, it can be used after installation, but it won't work after restarting!

**Comment by j-hc** [comment]
> > test this module: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb
> 
> It still doesn't work, it can be used after installation, but it won't work after restarting!

i dont care about "not working" comments. give a logcat or dont comment at all.

**Comment by ankit1w** [comment]
> test this module: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb

Same exception 
https://bin.kv2.dev/~647dc9a7e599576ea02c97cd

**Comment by TheNeutrinoRaged** [comment]
https://pastebin.com/PUma6PLc

**Comment by j-hc** [comment]
how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing

**Comment by ankit1w** [comment]
> how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing

Same exception

**Comment by j-hc** [comment]
> > how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing
> 
> Same exception

- remove module and reboot
- from inside the module install com.google.android.youtube.apk
- patch with revanced manager in root mode

**Comment by ankit1w** [comment]
> > > how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing
> > 
> > 
> > Same exception
> 
> * remove module and reboot
> * from inside the module install com.google.android.youtube.apk
> * patch with revanced manager in root mode
> 
> and see if it still crashes

Doesn't crash, works fine

**Comment by RobRoid** [comment]
> how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing

This actually fixed the problem, I used the newer module but It kept crashing on watching a video before it starts.
Tested on Samsung S10+

**Comment by j-hc** [comment]
> > how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing
> 
> This actually fixed the problem, I used the newer module but It kept crashing on watching a video before it starts. Tested on Samsung S10+

can you test this one too: https://drive.google.com/file/d/1kk0f4tgSjxwFLdxutuoOmU7w6jLd0I_b/view

do a clean installation (remove module, reboot, uninstall youtube, reflash the new one)

**Comment by TheNeutrinoRaged** [comment]
> can you test this one too: https://drive.google.com/file/d/1kk0f4tgSjxwFLdxutuoOmU7w6jLd0I_b/view
> 
> do a clean installation (remove module, reboot, uninstall youtube, reflash the new one)

still crashes for me https://gist.github.com/TheNeutrinoRaged/3e959e7b0886fb1735aa3b9986a8a4b0

**Comment by RobRoid** [comment]
> > > how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing
> > 
> > 
> > This actually fixed the problem, I used the newer module but It kept crashing on watching a video before it starts. Tested on Samsung S10+
> 
> can you test this one too: https://drive.google.com/file/d/1kk0f4tgSjxwFLdxutuoOmU7w6jLd0I_b/view
> 
> do a clean installation (remove module, reboot, uninstall youtube, reflash the new one)

Alright, will try soon.
I also noticed that the size became 97mb than the previous one, which is 153mb.

**Comment by RobRoid** [comment]
> > > > how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing
> > > 
> > > 
> > > This actually fixed the problem, I used the newer module but It kept crashing on watching a video before it starts. Tested on Samsung S10+
> > 
> > 
> > can you test this one too: https://drive.google.com/file/d/1kk0f4tgSjxwFLdxutuoOmU7w6jLd0I_b/view
> > do a clean installation (remove module, reboot, uninstall youtube, reflash the new one)
> 
> Alright, will try soon. I also noticed that the size became 97mb than the previous one, which is 153mb.

Apparently, YouTube had a new update 18.24.36
I think you need to update the module.

**Comment by j-hc** [comment]
> Apparently, YouTube had a new update 18.24.36 I think you need to update the module.

patches is not compatible yet.

try this too: https://drive.google.com/file/d/1tSGDiSi11Cu-KsMQEYppiJz5iHo0IP-f/view?usp=sharing

im just trying to be sure. i might need to rollback to using full stock apks which will increase the module size.

**Comment by RobRoid** [comment]
> > > how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing
> > 
> > 
> > This actually fixed the problem, I used the newer module but It kept crashing on watching a video before it starts. Tested on Samsung S10+
> 
> can you test this one too: https://drive.google.com/file/d/1kk0f4tgSjxwFLdxutuoOmU7w6jLd0I_b/view
> 
> do a clean installation (remove module, reboot, uninstall youtube, reflash the new one)

It works! I tried to watch a regular and shorts video, No issues.

**Comment by j-hc** [comment]
> > > > how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing
> > > 
> > > 
> > > This actually fixed the problem, I used the newer module but It kept crashing on watching a video before it starts. Tested on Samsung S10+
> > 
> > can you test this one too: https://drive.google.com/file/d/1kk0f4tgSjxwFLdxutuoOmU7w6jLd0I_b/view
> > 
> > do a clean installation (remove module, reboot, uninstall youtube, reflash the new one)
> 
> I had to downgrade the version to try this, and It works! I tried to watch a regular and shorts video, No issues.

are you sure? did you do a clean install?

**Comment by RobRoid** [comment]
> > > > > how about this: https://drive.google.com/file/d/1AT7UFbA6cREZWPzUPVSVlU1k706rEmXb/view?usp=sharing
> > > > 
> > > > 
> > > > This actually fixed the problem, I used the newer module but It kept crashing on watching a video before it starts. Tested on Samsung S10+
> > > 
> > > 
> > > can you test this one too: https://drive.google.com/file/d/1kk0f4tgSjxwFLdxutuoOmU7w6jLd0I_b/view
> > > do a clean installation (remove module, reboot, uninstall youtube, reflash the new one)
> > 
> > 
> > I had to downgrade the version to try this, and It works! I tried to watch a regular and shorts video, No issues.
> 
> are you sure? did you do a clean install?

Yes, I removed the module, rebooted, uninstalled youtube, rebooted again, reflashed the module, rebooted one more time, and opened youtube, and no issues.

**Comment by TheNeutrinoRaged** [comment]
> > Apparently, YouTube had a new update 18.24.36 I think you need to update the module.
> 
> patches is not compatible yet.
> 
> > still crashes for me https://gist.github.com/TheNeutrinoRaged/3e959e7b0886fb1735aa3b9986a8a4b0
> 
> try this too: https://drive.google.com/file/d/1tSGDiSi11Cu-KsMQEYppiJz5iHo0IP-f/view?usp=sharing
> 
> im just trying to be sure. i might need to rollback to using full stock apks which will increase the module size.

This one works for me, no more crashes!

**Comment by RobRoid** [comment]
> > Apparently, YouTube had a new update 18.24.36 I think you need to update the module.
> 
> patches is not compatible yet.
> 
> > still crashes for me https://gist.github.com/TheNeutrinoRaged/3e959e7b0886fb1735aa3b9986a8a4b0
> 
> try this too: https://drive.google.com/file/d/1tSGDiSi11Cu-KsMQEYppiJz5iHo0IP-f/view?usp=sharing
> 
> im just trying to be sure. i might need to rollback to using full stock apks which will increase the module size.

I think it's best to roll back to the original stock size because a week ago, I got the same issue as of today.

**Comment by j-hc** [comment]
fixed in next release a299cb0
